### PR TITLE
Added client-string generator

### PR DESCRIPTION
**Pull request by lazypower**
Adds a string builder to aid when handling *-relation-joined/changed to
    hand out a cluster connection string understood by both families of uses.

```
eg: docker expects etcd:// for swarm discovery eg: everything else on the
planet uses http:// or https:// urls
```
-  Adds icon, new interface declaration etcd-cluster
-  Import the legacy items from the original charm

### CODE REVIEW

**Comment by lazypower** [comment]
Related to work going on in https://github.com/chuckbutler/interface-etcd/pull/1

File: actions/health [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+PRIVATE_ADDRESS=$(unit-get private-address)
+OUT=$(/opt/etcd/etcdctl -C http://${PRIVATE_ADDRESS}:4001 cluster-health)
```
Comment (mbruzek) [inline actions/health:5]: fancy!

File: reactive/etcd.py [hunk 2]
```diff
@@ -68,6 +66,16 @@ def remove_unit_from_cluster(cluster):
     remove_state('cluster.departed')
 
 
+@when('db.connected')
```
Comment (mbruzek) [inline reactive/etcd.py:69]: Who sets this state?

Comment (mbruzek) [inline reactive/etcd.py:69]: interface is updated to set this state

File: tests/10-deploy [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    def test_two_node_scale(self):
+        self.deployment.add_unit('etcd')
+        self.deployment.sentry.wait()
+
+        status1 = self.deployment.sentry['etcd'][0].run('service etcd status')
+        status2 = self.deployment.sentry['etcd'][1].run('service etcd status')
+        self.assertTrue("running" in status1[0])
+        self.assertTrue("running" in status2[0])
+
```
Comment (mbruzek) [inline tests/10-deploy:33]: running != healthy cluster.  You could also run a `etcdctl cluster-health` on one of the machines here to make sure the cluster is healthy. Improvements for next iteration.

**Comment by mbruzek** [comment]
I have not tested this latest iteration but +1 LGTM

**Pull request merged**